File: src/lib/api/codeReview.ts
import { hasTauriRuntime, invoke } from "./client";
import type {
  CodeReviewBatchParseResult,
  CodeReviewTask,
  CreateCodeReviewBatchTasksInput,
  CreateCodeReviewTaskInput,
  ListCodeReviewTasksInput,
  ParseCodeReviewBatchInput,
  RunCodeReviewAiInput,
} from "@/types";

function requireTauriRuntime(): never {
  throw new Error("代码审核需要读取和操作本地 Git 工作区，请在 Tauri 桌面端使用该功能。");
}

export const codeReviewApi = {
  list: (input?: ListCodeReviewTasksInput) =>
    hasTauriRuntime()
      ? invoke<CodeReviewTask[]>("list_code_review_tasks", { input })
      : Promise.resolve([]),
  get: (taskKey: string) =>
    hasTauriRuntime()
      ? invoke<CodeReviewTask>("get_code_review_task", { taskKey })
      : Promise.resolve(requireTauriRuntime()),
  create: (input: CreateCodeReviewTaskInput) =>
    hasTauriRuntime()
      ? invoke<CodeReviewTask>("create_code_review_task", { input })
      : Promise.resolve(requireTauriRuntime()),
  createBatchTasks: (input: CreateCodeReviewBatchTasksInput) =>
    hasTauriRuntime()
      ? invoke<CodeReviewTask[]>("create_code_review_batch_tasks", { input })
      : Promise.resolve(requireTauriRuntime()),
  prepareDiff: (taskKey: string) =>
    hasTauriRuntime()
      ? invoke<CodeReviewTask>("prepare_code_review_diff", { taskKey })
      : Promise.resolve(requireTauriRuntime()),
  runAi: (input: RunCodeReviewAiInput) =>
    hasTauriRuntime()
      ? invoke<CodeReviewTask>("run_code_review_ai", { input })
      : Promise.resolve(requireTauriRuntime()),
  merge: (taskKey: string) =>
    hasTauriRuntime()
      ? invoke<CodeReviewTask>("merge_code_review_task", { taskKey })
      : Promise.resolve(requireTauriRuntime()),
  push: (taskKey: string) =>
    hasTauriRuntime()
      ? invoke<CodeReviewTask>("push_code_review_task", { taskKey })
      : Promise.resolve(requireTauriRuntime()),
  abortMerge: (taskKey: string) =>
    hasTauriRuntime()
      ? invoke<CodeReviewTask>("abort_code_review_merge", { taskKey })
      : Promise.resolve(requireTauriRuntime()),
  cancel: (taskKey: string) =>
    hasTauriRuntime()
      ? invoke<CodeReviewTask>("cancel_code_review_task", { taskKey })
      : Promise.resolve(requireTauriRuntime()),
  parseBatch: (input: ParseCodeReviewBatchInput) =>
    hasTauriRuntime()
      ? invoke<CodeReviewBatchParseResult>("parse_code_review_batch", { input })
      : Promise.resolve(requireTauriRuntime()),
};

File: src-tauri/src/commands/code_review.rs
use crate::error::CommandError;
use crate::models::{
    CodeReviewBatchParseResult, CodeReviewTask, CreateCodeReviewBatchTasksInput,
    CreateCodeReviewTaskInput, ListCodeReviewTasksInput, ParseCodeReviewBatchInput,
    RunCodeReviewAiInput,
};
use crate::services::code_review::CodeReviewService;
use crate::state::AppState;

#[tauri::command]
pub fn list_code_review_tasks(
    state: tauri::State<'_, AppState>,
    input: Option<ListCodeReviewTasksInput>,
) -> Result<Vec<CodeReviewTask>, CommandError> {
    CodeReviewService::list(&state.db, input).map_err(|e| e.into())
}

#[tauri::command]
pub fn get_code_review_task(
    state: tauri::State<'_, AppState>,
    task_key: String,
) -> Result<CodeReviewTask, CommandError> {
    CodeReviewService::get(&state.db, &task_key).map_err(|e| e.into())
}

#[tauri::command]
pub fn create_code_review_task(
    state: tauri::State<'_, AppState>,
    input: CreateCodeReviewTaskInput,
) -> Result<CodeReviewTask, CommandError> {
    CodeReviewService::create(&state.db, input).map_err(|e| e.into())
}

#[tauri::command]
pub fn create_code_review_batch_tasks(
    state: tauri::State<'_, AppState>,
    input: CreateCodeReviewBatchTasksInput,
) -> Result<Vec<CodeReviewTask>, CommandError> {
    CodeReviewService::create_batch_tasks(&state.db, input).map_err(|e| e.into())
}

#[tauri::command]
pub async fn prepare_code_review_diff(
    state: tauri::State<'_, AppState>,
    task_key: String,
) -> Result<CodeReviewTask, CommandError> {
    CodeReviewService::prepare_diff(&state.db, &task_key)
        .await
        .map_err(|e| e.into())
}

#[tauri::command]
pub async fn run_code_review_ai(
    state: tauri::State<'_, AppState>,
    input: RunCodeReviewAiInput,
) -> Result<CodeReviewTask, CommandError> {
    CodeReviewService::run_ai(&state.db, input)
        .await
        .map_err(|e| e.into())
}

#[tauri::command]
pub async fn merge_code_review_task(
    state: tauri::State<'_, AppState>,
    task_key: String,
) -> Result<CodeReviewTask, CommandError> {
    CodeReviewService::merge(&state.db, &task_key)
        .await
        .map_err(|e| e.into())
}

#[tauri::command]
pub async fn push_code_review_task(
    state: tauri::State<'_, AppState>,
    task_key: String,
) -> Result<CodeReviewTask, CommandError> {
    CodeReviewService::push(&state.db, &task_key)
        .await
        .map_err(|e| e.into())
}

#[tauri::command]
pub async fn abort_code_review_merge(
    state: tauri::State<'_, AppState>,
    task_key: String,
) -> Result<CodeReviewTask, CommandError> {
    CodeReviewService::abort_merge(&state.db, &task_key)
        .await
        .map_err(|e| e.into())
}

#[tauri::command]
pub fn cancel_code_review_task(
    state: tauri::State<'_, AppState>,
    task_key: String,
) -> Result<CodeReviewTask, CommandError> {
    CodeReviewService::cancel(&state.db, &task_key).map_err(|e| e.into())
}

#[tauri::command]
pub async fn parse_code_review_batch(
    state: tauri::State<'_, AppState>,
    input: ParseCodeReviewBatchInput,
) -> Result<CodeReviewBatchParseResult, CommandError> {
    CodeReviewService::parse_batch(&state.db, input)
        .await
        .map_err(|e| e.into())
}
